### PR TITLE
First draft of devkitPPC/wut setup instructions

### PR DESCRIPTION
**Pull request by CreeperMario**
Consider this to be a first draft, as there may be some factual inconsistencies. Specifically:

* I am unsure of the differences between WSL and standard GNU/Linux, and whether or not the build instructions would actually differ. I also have not considered specialised circumstances such as wanting to use WSL to perform the actual building of the program, but using a Windows-based text editor to modify source code
* I am unsure whether or not `devkitpro-pkgbuild-helpers` is strictly necessary, so I added it just in case
* Is `WUT_ROOT` exported automatically after installation? I think with `wut-linux` it is but `wut-linux-bin` and `wut-osx` (which are used in these instructions) do not export it
* `dkp-pacman` apparently provides scripts to automate the creation of environment variables, and I'm not quite sure how this works - e.g. is `DEVKITPPC` automatically exported upon installation?

### CODE REVIEW

**Comment by CreeperMario** [comment]
I just realised that I never mentioned anything about CMake being required - I'll add that shortly

File: tutorial/Chapter 1.md
Comment (ashquarky) [inline tutorial/Chapter 1.md]: > `devkitpro-pkgbuild-helpers` (helpful utilities for linking to software libraries).

Nah, that's not what they do at all! devkitPro uses them internally to compile all their packages. As far as I know, an average user shouldn't need them.

File: tutorial/Chapter 1.md
Comment (ashquarky) [inline tutorial/Chapter 1.md]: I'm not sure whether I'd prefer a direct link to README.md; thoughts?

File: tutorial/Chapter 1.md
Comment (ashquarky) [inline tutorial/Chapter 1.md]: `wut-linux` adds a file to /etc/profile that should do this automatically, though you're right in that neither `wut-linux-bin` or `wut-osx` do this. I could add it to `wut-linux-bin`, though honestly I have no idea how to approach it on OSX.

File: tutorial/Chapter 1.md
Comment (ashquarky) [inline tutorial/Chapter 1.md]: ehhhhhh
I get that `wut-linux-bin` is the official binaries, but I dunno...
Open to conversation on this one, because I honestly don't know what I prefer here.

File: tutorial/Chapter 1.md
Comment (ashquarky) [reply tutorial/Chapter 1.md]: quick note: devkitPro ships `devkit-env`, which does the same /etc/profile thing. IIRC it's not an explicit dependency of devkitPPC, so it should probably be added to the package list.

**Review by ashquarky — CHANGES_REQUESTED** [review]
I like it! Some minor tweaks in the review.
On a meta note, I've been working on rewriting the [wiiubrew tutorial](https://wiiubrew.org/wiki/Homebrew_development_guide) (draft [here](https://wiiubrew.org/wiki/User:QuarkTheAwesome/Dev_guide_draft) which basically covers all the same things, though I may just steal your text and use that instead - devkitPro's install "instructions" really aren't good enough. I don't see a problem with covering the same topic in both tutorials, just thought you should know. (editing that page is a-ok if you're in the mood ;D)
Can always tell when you're written something, your language is really verbose:
> you may wish to install Ubuntu or another GNU/Linux distribution from the Microsoft Store app, and perform these instructions within the virtualised Linux subsystem

It's not a problem, I just find it kinda funny. It's great!

File: tutorial/Chapter 1.md
Comment (CreeperMario) [reply tutorial/Chapter 1.md]: Fair enough, so just `devkitPPC` and `wiiload` then? I don't think anything else is necessary.

File: tutorial/Chapter 1.md
Comment (CreeperMario) [reply tutorial/Chapter 1.md]:
```suggestion
The process of setting up Fling is documented on its [GitLab repository page](https://gitlab.com/QuarkTheAwesome/wiiu-fling/blob/master/README.md#wiiu-fling).
```

File: tutorial/Chapter 1.md
Comment (CreeperMario) [reply tutorial/Chapter 1.md]: `devkit-env` is not in my list of packages. I looked at the [repository](https://github.com/devkitPro/tool-packages) and it does seem to install a script into `/etc/profile.d/` which adds `DEVKITPRO` and `DEVKITPPC` variables.

`/etc/profile.d` does not exist on my Mac, hence why I don't see `devkit-env` in my local package list. I'll see if there's an alternative way to programmatically add variables to shell sessions.

File: tutorial/Chapter 1.md
Comment (CreeperMario) [reply tutorial/Chapter 1.md]: I chose `wut-linux-bin` simply because it works in the same way as `wut-osx`.

Its your call though. 😁

File: tutorial/Chapter 1.md
Comment (CreeperMario) [reply tutorial/Chapter 1.md]: I found that on Linux, reading scripts from `/etc/profile.d/` is actually implemented in the shell script `/etc/profile` which does in fact also exist in macOS.

Hence, it is possible to "implement" this functionality into macOS by adding the following to the bottom of `/etc/profile` (modified from http://www.linuxfromscratch.org/blfs/view/6.3/postlfs/profile.html)
```shell
if [ -d /etc/profile.d ]; then
        for script in /etc/profile.d/*.sh; do
                if [ -x $script ]; then
                        . $script
                fi
        done
fi
```
I can confirm that this works on my system (the script I put within `/etc/profile.d` was run on my next terminal session), however it may not be the best idea to arbitrarily modify a system file like this.

In terms of stuff that is already built-in to macOS, the only similar functionality that I could find was `/etc/paths.d` which contains plain-text files with lists of directories that should be added to the `PATH` at the beginning of a shell session (MacGPG uses this to expose its implementation of things like `gpg` and `pinentry`). It looks like this directory is processed by `/usr/libexec/path_helper` which is called by default at the beginning of `/etc/profile`. It is useful for modifying `PATH`, but not for other variables such as `WUT_ROOT` though.

Comment (CreeperMario) [reply tutorial/Chapter 1.md]: Okay, so according to https://devkitpro.org/wiki/devkitPro_pacman#Customising_Existing_Pacman_Install -

If you're using a pre-existing msys2 install or OSX then please also set environment variables as follows:
```shell
DEVKITPRO=/opt/devkitpro
DEVKITARM=/opt/devkitpro/devkitARM
DEVKITPPC=/opt/devkitpro/devkitPPC
```

So, I think `wut-linux` and `wut-linux-bin` should add a file to `/etc/profile.d` which adds `WUT_ROOT` to the `PATH` automatically, but on macOS, much like with devkitPPC itself, the variables should be added to the user's `.bash_profile` manually.

File: tutorial/Chapter 1.md
Comment (ashquarky) [reply tutorial/Chapter 1.md]: I'm probably gonna delete one of them and rename the other to `wut-linux`, hoping exjam will have a preference there.

File: tutorial/Chapter 1.md
Comment (ashquarky) [reply tutorial/Chapter 1.md]: It seems that msys2 also has an `/etc/profile.d`, so Windows users *should* be able to use devkit-env (though for whatever reason devkitPro doesn't ship it). If we ever get a `wut-windows` - should be soon, the endian is fixed - then I'll probably drop something in /etc/profile.d.
As I mentioned elsewhere, I'm not sure how long `wut-linux-bin` is sticking around, but if I end up updating it I'll be sure to add an `/etc/profile.d` entry.

Not 100% on how to handle OSX, it may help to just have os-specific instructions.
> On Linux, install `devkit-env`. On Windows or OSX, run `echo EXPORT WUT_ROOT=... > ~/.bashrc`

Y'know? It'll at least work until devkitPro figures something out, then we can just copy them ;D

File: tutorial/Chapter 1.md
Comment (ashquarky) [reply tutorial/Chapter 1.md]: Cool, I'm happy with that! :+1:
(GitHub won't let me commit it, presumably I'd need write access to your fork. All good, commit it yourself and I'll merge everything in one batch)

File: tutorial/Chapter 1.md
Comment (ashquarky) [reply tutorial/Chapter 1.md]: Yeah, that should cover the basics. Windows/msys2 people can also install cmake through pacman, though I think that should get covered when installing wut.

**Review by ashquarky — APPROVED** [review]

**Pull request merged**
